pkgs/development/libraries/libav/default.nix

```diff
@@ -41,7 +41,10 @@ let
       inherit sha1; # upstream directly provides sha1 of releases over https
     };
 
-    patches = []
+    patches = [
+      # From Handbrake
+      ./A21-mp4-sdtp.patch
```

Comment:
Could you add a link to where the patch comes from, or even better use `fetchurl` to obtain it. Additionally, could you link to a libav issue tracking the merge of the patch into upstream libav?

Reply:
I searched and could not find a LibAV upstream issue about this patch.

Is it better to use URL and hash of `master` branch of upstream, or a URL or that commit tree?
In first case it would fail with hash error on any patch change.
On second case - it would work until it going to not.

Reply:
I could not find anything either. Asked about it on the PR: https://github.com/HandBrake/HandBrake/pull/1006

Without an upstream bug report, it definitely should not be applied to main libav package. Nixpkgs tries to be as close to upstream as possible to simplify maintenance, adding random patches is widely frowned upon.

You could use `(libav.overrideAttrs (super: {patches = (super.patches or []) ++ [(fetchurl {url = https://github.com/HandBrake/HandBrake/raw/7a49ab6e54a1fe6f33cf1e1ccbf46c8dd59015d1/contrib/ffmpeg/A21-mp4-sdtp.patch; sha256 = "xxx"; })];}))`, but personally, I find it more acceptable to be on Handbrake developers’ blacklist than have one more libav package to please abhorrent upstream policies.

Reply:
Or add patch to libav but disable it by default:

```nix
  patches = stdenv.lib.optionals withRandomHandbreakPatches [
    # taken from: https://github.com/HandBrake/HandBrake/pull/1006
    # upstream bug: xxx
    (fetchurl {
      url = https://github.com/HandBrake/HandBrake/raw/7a49ab6e54a1fe6f33cf1e1ccbf46c8dd59015d1/contrib/ffmpeg/A21-mp4-sdtp.patch;
      sha256 = "xxx";
    })
  ];
```

and use it with `(libav.override { withRandomHandbreakPatches = true; })`

Reply:
Unfortunately, not everything you see in nixpkgs is a good code. The patches in particular were often randomly added by someone in the past and nobody knows why were they added in the first place. I personally looked up and removed or added upstream links to almost a hundred patches.

Reply:
Cool.

It would be nice, if you can spare the time.